decentraland/helpers/index.js

```diff
@@ -178,6 +188,46 @@ export const openURL = async url => {
   }
 };
 
+export const restoreCreationStateOfAccountFromBlockchain = async account => {
+  const provider = ProviderFactory.getProvider();
+  const { address } = account;
+
+  const contracts = getContracts();
+  const { marketplaceContract, manaContract } = contracts;
+
+  const ethersBalance = await provider.getBalance(address);
+  const manaBalance = await manaContract.balanceOf(address);
+  const allowance = await manaContract.allowance(
+    address,
+    marketplaceContract.getAddress()
+  );
+
+  let creationStatus = NOT_STARTED;
+  let creationActions = {};
+
+  const accountCreated = account !== null;
+  const fundedWithEthers = `${ethersBalance}` !== `${ZERO}`;
+  const fundedWithMana = `${manaBalance}` !== `${ZERO}`;
+  const marketplaceWasApproved = `${allowance}` != `${ZERO}`;
+
+  if (accountCreated) {
+    creationStatus = FUNDING_WITH_ETH;
+    creationActions[FUNDING_WITH_ETH] = null;
+  }
+
+  if (fundedWithEthers) {
+    creationStatus = FUNDING_WITH_MANA_AND_APPROVING_MARKETPLACE;
+    creationActions[APPROVING_MARKETPLACE] = null;
+    creationActions[FUNDING_WITH_MANA] = null;
+  }
+
```

Comment:
Maybe handle the case here where either mana fanding or marketplace approve worked but not the other, given that it would just a few more lines, even if it's unlikely?

Reply:
This would be to set creationStatus, and we'd probably want an `else if` to handle this

Reply:
I've added the `APPROVING_MARKETPLACE` and `FUNDING_WITH_MANA` status, I haven't used the `else if`, I've tried to reproduce the same flow from `signup` using independent `ifs`

Reply:
Yeah true, the ordering of ifs can achieve the same effect as using else if

Reply:
Resolving this